### PR TITLE
TP/SL setting

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x] UpdateFormModal
- [x] TradeTab
- [x] TradeTabMobile

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/trade_tab/trade_tab.tsx [hunk 1]
```diff
@@ -370,6 +377,27 @@ const TradeTab = () => {
     );
   };
 
+  const setTpSlBounds = () => {
```
Comment (arealclimber) [inline src/components/trade_tab/trade_tab.tsx:380]: - 建議的TP/SL要在範圍內 `TradeTab` `TradeTabMobile` `UpdateFormModal`

Comment (arealclimber) [reply src/components/trade_tab/trade_tab.tsx:380]: - 計算強平價

File: src/components/trade_tab/trade_tab.tsx [hunk 2]
```diff
@@ -664,7 +746,8 @@ const TradeTab = () => {
   const displayedLongSlSetting = (
     <div className={`${isDisplayedLongSlSetting}`}>
       <TradingInput
-        lowerLimit={0}
+        lowerLimit={longSlLowerLimitRef.current}
+        upperLimit={longSlUpperLimitRef.current}
```
Comment (arealclimber) [inline src/components/trade_tab/trade_tab.tsx:750]: - 推薦強平價就不能+-

Comment (arealclimber) [reply src/components/trade_tab/trade_tab.tsx:750]: - incrementButton in `TradingInput`

File: src/components/update_form_modal/update_form_modal.tsx [hunk 3]
```diff
@@ -67,35 +73,53 @@ const UpdateFormModal = ({
...
   const [guaranteedStopFee, setGuaranteedStopFee, guaranteedStopFeeRef] = useStateRef(
     Number(gsl) * openCfdDetails?.openPrice * openCfdDetails?.amount
   );
 
-  useEffect(() => {
-    setGuaranteedStopFee(Number(gsl) * openCfdDetails?.openPrice * openCfdDetails?.amount);
-  }, [gsl, openCfdDetails?.openPrice, openCfdDetails?.amount]);
+  const [disableSlInput, setDisableSlInput, disableSlInputRef] = useStateRef(false);
+
+  const [caledSlLowerLimit1, caledSlUpperLimit1] = useMemo(() => {
```
Comment (arealclimber) [inline src/components/update_form_modal/update_form_modal.tsx:98]: - 刪掉

File: src/constants/config.ts [hunk 4]
```diff
@@ -266,3 +266,8 @@ export const TERM_OF_SERVICE = 'https://www.tidebit-defi.com/term_of_service/{ha
 export const PRIVATE_POLICY = 'https://www.tidebit-defi.com/private_policy/{hash}';
 export const DeWT_VALIDITY_PERIOD = 60 * 60; // seconds
 export const CFD_LIQUIDATION_TIME = 86400;
+export const SELL_PRICE_ERROR = 0;
+export const BUY_PRICE_ERROR = 9999999999;
+export const LEVERAGE_ERROR = 1;
+export const USER_BALANCE_ERROR = 0;
```
Comment (arealclimber) [inline src/constants/config.ts:272]: 應該用 DEFAULT_ 來命名，並且應該移到display

```
export const SELL_PRICE_ERROR = 0;
export const BUY_PRICE_ERROR = 9999999999;
export const LEVERAGE_ERROR = 1;
export const USER_BALANCE_ERROR = 0;
```

File: src/constants/config.ts [hunk 5]
```diff
@@ -266,3 +266,8 @@ export const TERM_OF_SERVICE = 'https://www.tidebit-defi.com/term_of_service/{ha
 export const PRIVATE_POLICY = 'https://www.tidebit-defi.com/private_policy/{hash}';
 export const DeWT_VALIDITY_PERIOD = 60 * 60; // seconds
 export const CFD_LIQUIDATION_TIME = 86400;
+export const SELL_PRICE_ERROR = 0;
```
Comment (Luphia) [inline src/constants/config.ts:269]: - remove and rename these constant params
  - SELL_PRICE_ERROR -> DEFAULT_SELL_PRICE
  - BUY_PRICE_ERROR -> DEFAULT_BUY_PRICE
  - USER_BALANCE_ERROR -> DEFAULT_BALANCE_AMOUNT

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**